### ISSUE TITLE
pbojinov / request-ip vs supercharge/request-ip

### ISSUE THREAD

**Issue by evdama** [supercharge/request-ip#1]
Hi Marcus,

I came here from https://github.com/pbojinov/request-ip/pull/33

I'd be willing to switch my app to your package.
Can you maybe add a bit info, maybe even to the README as I assume other will look for the same inital bits of information:
 - do you have feature parity with petars request-ip?
 - do you have any fixes/additions in addition to his package?
 - what else? why would one want to switch or not?

Thanks much for the info 👍

**Comment by evdama** [comment]
Trying to run with this package in my Sapper project I get this error when I try using it in my `server.js` file:

```
✗ server
'getClientIp' is not exported by node_modules/@supercharge/request-ip/index.js, imported by src/server.js
 6: import 'nprogress/nprogress.css'
 7: import { callToIPRegistryAPI } from 'utils'
 8: import { getClientIp } from '@supercharge/request-ip'
             ^
```
trying to use it like so
```js
import { getClientIp } from '@supercharge/request-ip'
...
request.session.clientIPAddress.ip = await getClientIp( request )
```

**Comment by marcuspoehls** [comment]
@evdama Hey Markus, thank you for your questions.

**Feature Parity: yes**
This package has all features that Petar’s `request-ip` package has. I’m aiming to make it as seamless as possible to switch.


**Additional Features**
I added the functionality to retrieve the request’s IP address from `request.raw`. This is helpful when using Fastify as your web framework. Otherwise, this package supports the same request locations as the one from Petar.


**Why making the switch?**
I’m actively maintaining the package. I’ll work through feature requests and bug reports. Petar’s package looks abandoned. With an active community around my hapi.js plugins, I want to support different setups. Creating my own package also gives me ownership of the tool to simply extend it when needed. For example, if a service like Cloudflare introduces a new location for the request’s IP address, I can quickly add it to the package.

**Comment by marcuspoehls** [comment]
@evdama I’ve released version `1.1.0` to address ES module support. If that doesn’t fix your issues, please let me know

**Comment by evdama** [comment]
I'm still getting an error on v 1.1.0
```
[!] Error: 'getClientIp' is not exported by node_modules/@supercharge/request-ip/index.js, imported by src/server.js
https://rollupjs.org/guide/en/#error-name-is-not-exported-by-module
/Users/sa/0/edm/src/server.js (8:9)
 6: import 'nprogress/nprogress.css'
 7: import { callToIPRegistryAPI } from 'utils'
 8: import { getClientIp } from '@supercharge/request-ip'
             ^
```

**Comment by marcuspoehls** [comment]
Ok, I’m thinking of another issue (possibly related to rollup bundling this package). I’ve published version `1.1.1` addressing the exports and making them explicit.

@evdama Can you please test against version `1.1.1`?

**Comment by marcuspoehls** [comment]
@evdama Hey Markus, I’m this issue here and assume it’s working for you now. If not, please don’t hesitate to reopen this issue. Thank you!
